### PR TITLE
Auto add contacts bug fixes tests

### PR DESCRIPTION
**Pull request by dangood84**
Hey gents,
Just added some tests / lines of code to test for the original bug fixes that were fixed as part of the development for the Trust Chair and School Chair contacts stuff

1. Checking we can save existing contacts from something back to null string
2. Checking that the error summary link for incorrectly entering numbers has the correct href attributes and values

### CODE REVIEW

**Review by sukhybhullar-nimble — APPROVED** [review]

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects.CypressTests/cypress/e2e/project-contacts.cy.ts [hunk 1]
```diff
@@ -102,6 +106,7 @@ describe("Testing that we can add contacts", () => {
                 .errorForTrustChairName("Trust chair name must not include special characters other than , ( )")
                 .withTrustChairName("4da")
                 .clickContinue()
+                .hasCorrectErrorSummaryLink()
```
Comment (danielryannimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects.CypressTests/cypress/e2e/project-contacts.cy.ts:109]: what is this testing for?...if we are testing error summary links why isnt there one for every error scenario?

Comment (danielryannimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects.CypressTests/cypress/e2e/project-contacts.cy.ts:109]: not sure if this is necessary.

Comment (dangood84) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects.CypressTests/cypress/e2e/project-contacts.cy.ts:109]: Dropped in to cover original bug 154613 - Some Errors Don't Link Down and miss respective validation error above the control

That was the particular error where the problem occured 👍

Comment (dangood84) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects.CypressTests/cypress/e2e/project-contacts.cy.ts:109]: @sukhybhullar-nimble - What are your thoughts on this?

I like the idea of covering ourselves with tests from bugs originally caught when developing the feature but is @danielryannimble right with this being too much effort here?

Comment (danielryannimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects.CypressTests/cypress/e2e/project-contacts.cy.ts:109]: we shouldn't a write a single scenario for a bug that once existed.....if we are testing for correct error summary links then we should test them all .. in every test

Comment (sukhybhullar-nimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects.CypressTests/cypress/e2e/project-contacts.cy.ts:109]: As discussed in a call "showsError" covers this link checking

**Review by danielryannimble — APPROVED** [review]

**Pull request merged**